frontend/src/components/MonacoEditor.tsx

```diff
@@ -134,7 +135,7 @@ const MonacoEditor = ({ content, filename }: Props) => {
       </div>
       <div style={{ fontSize: 14, marginTop: 5, marginBottom: 5 }}>
         {(!user || !user.me) && 'Please login to enable saving'}
-        {user?.me && currentBranch && `On branch ${currentBranch}`}
+        {commitMessage && `Last commit: ${commitMessage}`}
```

Comment:
Would "Latest commit" be more descriptive?
Also when the user is not logged in, the displayed message looks like this:
![img](https://i.imgur.com/9UJDQsL.png)
Maybe the "Please login to enable saving" is not required anymore because the modal prevents the user from saving anyways.